src/Spanner/Session/CacheSessionPool.php

```diff
@@ -777,4 +764,63 @@ private function validateConfig()
             );
         }
     }
+
+    /**
+     * Delete the provided sessions.
+     *
+     * @param array $sessions
+     */
+    private function deleteSessions(array $sessions)
+    {
+        $calls = [];
+
+        foreach ($sessions as $session) {
+            $calls[] = $this->database->connection()
+                ->deleteSessionAsync([
+                    'name' => $session['name'],
+                    'database' => $this->database->name()
+                ]);
+        }
+
+        foreach ($calls as $call) {
+            $call->wait();
```

Comment:
@michaelbausor, I haven't had any luck with getting the calls to complete without calling wait. 😕 

What if we set the lowest possible [timeout](https://github.com/grpc/grpc/blob/master/src/php/lib/Grpc/AbstractCall.php#L51) for each delete request? That could help minimize the effect of requests that may for some reason take longer than expected and we wouldn't need to worry about registering a shutdown function.